src/Application.php

```diff
@@ -639,33 +469,9 @@ private function checkForDuplicateRoute($path, $methods = null)
      */
     private function emitMarshalServerRequestException($exception)
     {
-        $response = (new Response())
+        $response = $this->responsePrototype
             ->withStatus(StatusCode::STATUS_BAD_REQUEST);
-        $finalHandler = $this->getFinalHandler();
-        $response = $finalHandler(new ServerRequest(), $response, $exception);
```

Comment:
I went a step further, and updated the method as follows:

- Checks to see if a container is present, and if it contains a `Zend\Expressive\Middleware\ErrorResponseGenerator` service; if not, it uses the previous logic.
- If it does, it pulls the service.
- It then invokes it, using the exception, an empty request, and the response prototype, in order to obtain a response.
- Logic then proceeds as normal for emitting the response.

This will provide a better error page for users. If they are using the whoops-based error generator, it will generate a whoops response with the exception details. If they are using the templated error generator, it will generate a templated response, with exception details if the debug flag is enabled.

Thanks for the heads-up and clarifying what you were flagging!